601 überarbeitung vitepress doku

MrSebastian opened the pull request:
# Beschreibung:

- Im wesentlichen wurden Elemente neu angeordnet, und Absätze verschoben, auch auf andere Seite
- Die Beschreibung der Workflows wurde umfangreicher überarbeitet
- Ich habe versucht alle Typos zu finden
- eine Seite zu Contribution habe ich nicht erstellt (war Optional)

## Definition of Done (DoD):
<!-- Je nach Service bitte nicht relevanten Teil entfernen-->

<!-- Dokumentation -->
### Dokumentation
- [x] Links geprüft

# Referenzen[^1]:

Verwandt mit Issue #

Closes #601

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

# Release Notes

- **Documentation**
  - Updated documentation structure and navigation
  - Added new sections for Ecosystem, Tools & Frameworks, and Workflows
  - Improved naming conventions and guidelines for service development

- **New Features**
  - Added "Getting Started" action on homepage
  - Introduced comprehensive workflow documentation
  - Created new guides for microservice creation

- **Improvements**
  - Refined configuration for documentation site
  - Enhanced search and localization options
  - Corrected grammatical and formatting issues across multiple documents

- **Chores**
  - Reorganized technical documentation
  - Updated sidebar and navigation structure
  - Removed outdated sections and consolidated information

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

vjohnslhm committed:
#601 delete placeholder on homepage

vjohnslhm committed:
#601 list sidebar elements alphabetically

vjohnslhm committed:
#601 translate default ui elements to german

MrSebastian committed:
Merge branch 'dev' into 601-überarbeitung-vitepress-doku

MrSebastian committed:
redesign coding -> naming conventions

MrSebastian committed:
moved naming conventions for workflow

MrSebastian committed:
replace general tips and tricks with its only existing db tip

MrSebastian committed:
created page set for ecosystem

MrSebastian committed:
added formatter setup description

was in get_started before

MrSebastian committed:
init page to setup a new microservice

MrSebastian committed:
new linked getting started page

MrSebastian committed:
rename adr navigation text

MrSebastian committed:
improve paragraph for set up database

MrSebastian committed:
add link for system user for oracle access

MrSebastian committed:
describe actions to create a new service

MrSebastian committed:
fixing typos

MrSebastian committed:
redesign workflow description and add sequence diagrams

MrSebastian committed:
change to short description of page set

MrSebastian committed:
reorder navigation items in alphabetic order

MrSebastian commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+
+Dabei sollte auf folgendes Schema geachtet werden:
+
+- Benutzername: \<Name des Services mit Unterstrichen\>
+- Passwort: secret
+
+Beispiel für `wls-broadcast-service`:
+- Benutzername: `wls_broadcast_service`
+- Passwort: `secret`
```
Sind Testdaten die nur bei lokaler Entwicklung vorhanden sind

vjohnslhm commented on docs/src/services/index.md:
hier muss die überschrift "Features" noch in "Services" umbenannt werden, und das `🚧 in Progress` kann entfernt werden, wenn es kein offenes Issue mehr gibt, in welchem das behandelt wird.

MrSebastian replied:
links auf Issues ergänzt: 67877ee9d892cc25aeb3cc0d1ce5944fc97f4ba8

vjohnslhm commented on docs/.vitepress/config.mts:
```diff
@@ -1,7 +1,8 @@
 const PATH_TECHNIK = '/technik/';
-const PATH_CODING_CONVENTIONS = PATH_TECHNIK + 'coding_conventions/';
+const PATH_ECOSYSTEM = PATH_TECHNIK + '/ecosystem/';
+const PATH_CODING_CONVENTIONS = PATH_TECHNIK + 'naming_conventions/';
```
```suggestion
const PATH_NAMING_CONVENTIONS = PATH_TECHNIK + 'naming_conventions/';
```

MrSebastian replied:
fixed mit 512757fe1acf6f872c14b41e715da969be540652

vjohnslhm commented on docs/src/technik/get_started/index.md:
```diff
@@ -6,39 +6,101 @@ Im Projekt verwenden wir `checkstyle` und `spotless` um für einen möglichst ei
+## Datenbank
 
-`wls-broadcast-service_push-dev-yml` ist der Workflow zum `wls-broadcast-service` der bei einem `push` auf `dev` ausgeführt wird.
+Der Zugriff auf die Oracle-Datenbank ist [einzurichten](/technik/guides/db-access).
```
```suggestion
Der Zugriff auf die Oracle-Datenbank über die IDE ist gemäß [dieser Anleitung](/technik/guides/db-access) einzurichten.
```

vjohnslhm commented on docs/src/services/index.md:
```diff
@@ -7,7 +7,7 @@ Für jeden Service gibt es eine separate Beschreibung zu dessen Aufgaben und Fun
-Kommunal- und/oder Landes- Wahlrechts erstellt.
+Kommunal- und/oder Landeswahlrechts erstellt.
 
 ![Datenmodell:](/fachlichesDatenmodell/vermutetesFachlichesDatenmodell_20241008.png)
```
sollen wir dieses datenmodell ggf auch in mermaid schreiben, damit es bessere qualität hat?

MrSebastian replied:
Das könnte mit der Größe Probleme machen, vor allem wenn noch mehr hinzukommt. Das Bild kann man in einem neuen Tab öffnen und anschließend mit den Browsermitteln zoomen.

Ich würde es mir gesondert ansehen und wenn es gut erscheint Issue und PR erstellen.

vjohnslhm commented on docs/src/technik/ecosystem/workflows.md:
```diff
@@ -0,0 +1,131 @@
+
+## Releasing
+
+Im Rahmen des Releasing werden getestet Versionen zur Verfügung gestellt.
```
```suggestion
Im Rahmen des Releasing werden getestete Versionen zur Verfügung gestellt.
```

vjohnslhm commented on docs/src/technik/naming_conventions/workflows.md:
```diff
@@ -0,0 +1,24 @@
+# Naming Convetions für Workflows
+
+❗ Ist zu beachten dass alle Workflows im Ordner `workflows` liegen müssen. Sie dürfen nicht auf unterordner aufgeteilt werden.
```
```suggestion
❗ Es ist zu beachten dass alle Workflows im Ordner `workflows` liegen müssen. Sie dürfen nicht auf unterordner aufgeteilt werden.
```

MrSebastian replied:
fixed mit b920e3150672adbcf64c79767555b320c5492a13

Ich habe den Bereich auch auf einen Alert umgestellt: 75b111b2194f0483b5a4ec7c22319fc9c814c8a1

vjohnslhm commented on docs/src/technik/naming_conventions/workflows.md:
```diff
@@ -0,0 +1,24 @@
+
+- `callable-<kurzbeschreibung>.yml` ... [wiederverwendbare Workflows](https://docs.github.com/de/actions/using-workflows/reusing-workflows)
+- `dispatch-<kurzbeschreibung>.yml` ... [manuell ausführbare workflows](https://docs.github.com/de/actions/using-workflows/manually-running-a-workflow)
+- `<service-namer>_<trigger>.yml` ... Workflows zu Jobs die durch ein bestimmtes Ereignis getriggert werden.
```
```suggestion
- `<service-name>_<trigger>.yml` ... Workflows zu Jobs die durch ein bestimmtes Ereignis getriggert werden.
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+## Datenbank einrichten
+
+Jeder Service bekommt einen eigenen Benutzer für die Datenbank. Die Zugriffs-URL ist für alle Services gleich:
+`jdbc:oracle:thin:@//localhost:1521/XEPDB1`
```
der hinweis 
> Neben dem Standardbenutzer der auf alles zugreifen kann (siehe docker-compose.yml) müssen alle weiteren Benutzer über `stack/add-user-on-startup.sql` erstellt werden.

ist jetzt weg, weil der zugriff über den Standardbenutzer nicht funktioniert? (war zumindest bei mir mal so)

```suggestion
## Datenbank einrichten

Jeder Service hat einen eigenen Benutzer für die Datenbank. Diese sind im File `stack/oracle-database/add-user-on-startup.sql` hinterlegt. Die Zugriffs-URL ist für alle Services gleich:
`jdbc:oracle:thin:@//localhost:1521/XEPDB1`
```

vjohnslhm replied:
wenn der standardnutzer noch aktuell ist, würde ich es hier in einem kurzen satz noch erwähnen, und auf `technik/guides/db-access` verweisen

vjohnslhm commented on docs/.vitepress/config.mts:
```diff
@@ -80,19 +66,47 @@ export default withMermaid({
+                            link: `${PATH_GUIDES}how-to-create-client-from-open-api-json.md`
+                        },
+                        {text: 'Datenbankzugriff', link: `${PATH_GUIDES}db-access.md`},
+                        {text: 'Neuer Mikroservice', link: `${PATH_GUIDES}new-service.md`}
```
```suggestion
                        {text: 'Neuer Microservice', link: `${PATH_GUIDES}new-service.md`}
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+
+## Maven-Projekt anlegen
+
+Für den neuen Service wird ein Ordner parallel zu den anderen Service angelegt. Dabei ist auf das Namensschema zu achten:
```
```suggestion
Für den neuen Service wird ein Ordner parallel zu den anderen Services angelegt. Dabei ist auf das Namensschema zu achten:
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+Für den neuen Service wird ein Ordner parallel zu den anderen Service angelegt. Dabei ist auf das Namensschema zu achten:
+`wls-<Domain>-service`
+
+In dem Ordner wird das Maven-Projekt eingerichtet. Dazu aus den [RefArch-Templates](https://github.com/it-at-m/refarch-templates)
```
ich würde konkret auf das backend-template verweisen, weil die anderen punkte auf dieser seite sich ausschließlich auf backend-spezifische konfigurationen beziehen. ggf kann man die überschrift anpassen, dass es sich um die erstellung eines backend-microservices handelt.
```suggestion
In dem Ordner wird das Maven-Projekt eingerichtet. Dazu aus den [RefArch-Templates](https://github.com/it-at-m/refarch-templates/tree/main/refarch-backend)
```

MrSebastian replied:
Link angepasst; den Bereich auf Backendend umgruppiert: 4453cf784c6e55730cda94b68cad34cdda21f1e2

Issue für Frontend erstellt: f7719698a30c291a0cbe85469a7ced4c0187328c

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+
+### Update der Tests
+
+Im Projekte haben wir für das Naming unserer Tests [Konventionen](/technik/naming_conventions/testing) aufgestellt. Die bereitgestellten Tests des Templates
```
```suggestion
Im Projekt haben wir für das Naming unserer Tests [Konventionen](/technik/naming_conventions/testing) aufgestellt. Die bereitgestellten Tests des Templates
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+Im Projekte haben wir für das Naming unserer Tests [Konventionen](/technik/naming_conventions/testing) aufgestellt. Die bereitgestellten Tests des Templates
+müssen entsprechend angepasst werden.
+
+## Workflows einrichten
```
hier könnte man ein beispiel einfügen, an dem man sehen kann, wo beim kopieren änderungen gemacht werden müssen:

```yml
name: verify pull request <<domain>>-service

on:
  pull_request:
    paths:
      - 'wls-<<domain>>-service/**'
      - '.github/workflows/wls-<<domain>>-service_pull-request.yml'

jobs:
  verify-pull-request:
    uses:
      ./.github/workflows/callable-run-mvn-verify.yml
    with:
      pom-dir: 'wls-<<domain>>-service'
```

vjohnslhm replied:
oder auch die zwei oder drei workflow files auflisten, die geändert/kopiert werden müssen, damit keins vergessen wird.

MrSebastian replied:
die beiden Workflows als Beispiel ergänzt: f0e272853e6706a200d2f7a21fd28218da5b4731

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+- Benutzername: `wls_broadcast_service`
+- Passwort: `secret`
+
+# Routing im Gateway einrichten
```
```suggestion
## Routing im Gateway einrichten
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+
+Anfragen die anden Broadcast-Service gehen sollen beginnen im Path mit `/api/broadcast-service/`.
+
+# Pflege der Rechte im Auth-Service
```
```suggestion
## Pflege der Rechte im Auth-Service
```

vjohnslhm commented on docs/src/technik/guides/new-service.md:
```diff
@@ -0,0 +1,64 @@
+
+Beispiel:
+
+Anfragen die anden Broadcast-Service gehen sollen beginnen im Path mit `/api/broadcast-service/`.
```
```suggestion
Anfragen die an den Broadcast-Service gehen sollen beginnen im Path mit `/api/broadcast-service/`.
```

vjohnslhm commented on docs/src/technik/index.md:
```diff
@@ -1,22 +1,2 @@
-- VitePress
\ No newline at end of file
+Hier sind alle Informationen zum Ökosystem, Konventionen, Designentscheidungen und was für den Beginn der Entwicklung wichtig ist,
+zusammengetragen.
```
ich würde die überschrift lassen, weil die seite sonst sehr leer aussieht, oder alternativ im `config.mts` file direkt auf getting-started verlinken, wenn man auf technik klickt.

```suggestion
# Technik vom WLS

Hier sind alle Informationen zum Ökosystem, Konventionen, Designentscheidungen und was für den Beginn der Entwicklung wichtig ist,
zusammengetragen.
```

MrSebastian replied:
auf getting started verlinkt: 853eb9ca9fb7446a49a9d5c7187d85f3f8672989

vjohnslhm commented on docs/src/services/ergebnismeldung-service/index.md:
```diff
@@ -37,7 +37,7 @@ Die Methode initialisiereAWerte() initialisiert die Wahlberechtigten (A-Werte) f
 - Für jede Wahlbezirk-ID in der Liste wird über die EAI im externen Wahlsystem nach A-Werten gesucht. Bei Erfolg werden
 diese im lokalen Repository gespeichert.
-- Bei Nichterfolg, wird geprüft ob im lokalen Repository wenigstens 'alte' A-Werte existieren.
+- Bei Nichterfolg, wird geprüft, ob im lokalen Repository wenigstens 'alte' A-Werte existieren.
```
```suggestion
- Bei Nichterfolg wird geprüft, ob im lokalen Repository wenigstens 'alte' A-Werte existieren.
```

vjohnslhm commented on docs/src/technik/adr/adr-improve-getting-started-of-services.md:
dieser adr ist nicht mehr aktuell, weil sich die ports geändert haben. hier sollten die informationen entsprechend angepasst werden.

MrSebastian replied:
ich habe notiz mit Link hinterlegt: 33f7c57b458f3029b0e2f6da47a937f9ac21947c

Aufgrund von Mermaid oberhalb der Ports wird man aber nicht direkt an der richtigen Stelle landen

vjohnslhm commented on docs/src/technik/ecosystem/workflows.md:
```diff
@@ -0,0 +1,131 @@
+# Workflows
+
+Mit [Workflows](https://docs.github.com/en/actions/writing-workflows) automatisieren wir Prozessen in der Entwicklung.
```
```suggestion
Mit [Workflows](https://docs.github.com/en/actions/writing-workflows) automatisieren wir Prozesse in der Entwicklung.
```

vjohnslhm commented on docs/src/technik/ecosystem/workflows.md:
```diff
@@ -0,0 +1,131 @@
+
+## Pull-Requests
+
+Mit der Erstellung eines Pull-Requests wird geprüft das der Code den Anforderungen und den Richtlinien entspricht.
```
```suggestion
Mit der Erstellung eines Pull-Requests wird geprüft, dass der Code den Anforderungen und den Richtlinien entspricht.
```

vjohnslhm commented on docs/src/technik/ecosystem/workflows.md:
```diff
@@ -0,0 +1,131 @@
+
+#### wls-common
+
+- Maven-Artefakte: https://central.sonatype.com/search?q=de.muenchen.oss.wahllokalsystem&sort=published
+- Github-Tag: https://github.com/it-at-m/Wahllokalsystem/tags
+- Github-Release: https://github.com/it-at-m/Wahllokalsystem/releases?q=wls-common&expanded=true
+
+#### Mikroservice
+
+- Github-Tag: https://github.com/it-at-m/Wahllokalsystem/tags
+- Github-Release: https://github.com/it-at-m/Wahllokalsystem/releases
+  - bei Backend-Services wird die openAPI-Spezifikation den Assets beigelegt 
+- Github-Imageregistry: https://github.com/orgs/it-at-m/packages?repo_name=Wahllokalsystem
+  - das neue Image hat den Tag `latest` und bei Backend-Services zusätzliche Tags entsprechend der Version
```
ich fände es schöner, wenn wir links alle als `[beschreibung](link)` handhaben und nicht einfach reinkopieren.

vjohnslhm replied:
so könnten die links auch wiederverwendet werden, wenn sie öfter gebraucht werden:
```
// im text
[Github-Tag][github-tag-link]

// ganz unten im file
[github-tag-link]: https://github.com/it-at-m/Wahllokalsystem/tags
```

MrSebastian replied:
👍 done with 47b1ac3ec80669299d22bf2c72b423cea6be2afb

vjohnslhm commented on docs/src/technik/get_started/index.md:
```diff
@@ -6,39 +6,101 @@ Im Projekt verwenden wir `checkstyle` und `spotless` um für einen möglichst ei
+| [Infomanagement](/services/infomanagement-service/)                      | 8201 |
+| [Monitoring](/services/monitoring-service/)                              | 8206 |
+| [Vorfälle und Vorkommnisse](/services/vorfaelleundvorkommnisse-service/) | 8204 |
+| Wahlvorbereitung                                                         | 8203 |
```
dass die doku für den broadcast service fehlt, ist bereits im issue #544 festgehalten. ich denke wir sollten auch ein issue für die wahlvorbereitung anlegen.

MrSebastian replied:
Issue erstellt: #739

vjohnslhm reviewed (changes requested):
1. alle seiten aus der sidebar haben ein top-level `index.md` file, außer die `ecosystem` seite. das würde ich in die eine oder in die andere richtung vereinheitlichen.
2. bei den adrs gibt es noch die seite `vollständige migration nach keycloak` die kann glaube ich gelöscht werden.
3.  ich würde überall wo wir "im aufbau" oder 🚧 stehen haben ein entsprechendes issue dazu aufmachen und in der doku auf das issue verweisen, oder den bearbeitungsstatus aus der doku entfernen. 
4. es gibt noch einen bug, dass manchmal die seitennavigation mit previous und next site nicht korrekt funktioniert. zum beispiel hier:
![grafik](https://github.com/user-attachments/assets/3c2dc4f7-5338-4699-bdf5-800e4aa6e0b5)
das kann aber in einem neuen issue ausgelagert werden, weil es kein "redesign" ist, wie die anderen sachen in diesem pr.

vjohnslhm commented:
@coderabbitai findest du in den dateien dieses pull request noch irgend welche rechtschreib- oder grammatikfehler?

MrSebastian committed:
👌 reviewfeedback - fix typos

Co-authored-by: Viviane Johns <145964798+vjohnslhm@users.noreply.github.com>

MrSebastian committed:
change level 1 heading

MrSebastian committed:
remove in progress

MrSebastian committed:
remove work in progress sentence

MrSebastian committed:
replace work in progress text with link to issue

MrSebastian committed:
match constant name to its content

MrSebastian committed:
replace ❗ with github-flavored alerts

MrSebastian committed:
fix typo

MrSebastian committed:
Anpassung der Gliederung auf Backend-Microservice

MrSebastian committed:
add heading with link to issue for frontend

MrSebastian committed:
add copy and paste template for workflows

MrSebastian committed:
use getting started as landing for technik

MrSebastian committed:
reuse links

MrSebastian committed:
fix typos

MrSebastian committed:
Unify authority strings

MrSebastian committed:
fix typo

MrSebastian committed:
add note for ports

vjohnslhm commented on docs/src/services/index.md:
```diff
@@ -1,23 +1,21 @@
-# Features
-
-🚧 in Progress
+# Service
```
```suggestion
# Services
```

MrSebastian replied:
fixed mit 785d9dd36b22ee03da2b817e2e7661265250c82b

vjohnslhm commented on docs/src/technik/ecosystem/workflows.md:
```diff
@@ -0,0 +1,134 @@
+- [Github-Tag][link-github-tag]
+- [Github-Release][link-github-release]
+  - bei Backend-Services wird die openAPI-Spezifikation den Assets beigelegt 
+- [Github-Imageregistry](https://github.com/orgs/it-at-m/packages?repo_name=Wahllokalsystem)
```
der link für die Github-Imageregistry ist auch doppelt drin (Zeile 88) und kann ausgelagert werden

MrSebastian replied:
fixed mit db45063a9802c33f56c9ceeb1a844e8b3185289b

vjohnslhm commented on docs/src/technik/naming_conventions/workflows.md:
```diff
@@ -0,0 +1,25 @@
+
+`dispatch-microservice-mvn-release.yml` ist ein Workflow, der manuell getriggert wird um ein Maven-Release eines Mikroservices durchzuführen.
+
+### `<service-namer>_<trigger>.yml`
```
ist das `r` in `service-namer` absicht?

MrSebastian replied:
fixed mit 79249e270d07546191b4e922b82c943ca0f5ef34

MrSebastian committed:
Merge branch 'dev' into 601-überarbeitung-vitepress-doku

MrSebastian committed:
fix formatting and indent

MrSebastian committed:
fix navigation issue

naming in ecosystem was bugged. navigation at end of page navigated to get startet

MrSebastian committed:
add description for ecosystem pages

MrSebastian commented:
@vjohnslhm Das Problem mit der Navigation konnte ich lösen. Manchmal ist es nur 1 Zeichen: 9747acc904ad8c63f150beb8ca7c373dc7452154

MrSebastian committed:
Merge remote-tracking branch 'origin/601-überarbeitung-vitepress-doku' into 601-überarbeitung-vitepress-doku

MrSebastian committed:
fix broken path for guides in navigation

MrSebastian committed:
Merge branch 'dev' into 601-überarbeitung-vitepress-doku

MrSebastian committed:
fix typo

MrSebastian committed:
use reference links for github image registry

MrSebastian committed:
fix typo
